indicator-nara/031-berekening.Rmd

```diff
@@ -0,0 +1,707 @@
+Deze kritische drempelwaarde is specifiek voor elk meetpunt.
+Als kritische drempelwaarde wordt gekozen voor 
+**het _jaargemiddelde_ van de 5%-percentiel van de grondwatermetingen _van een jaar_ voor het meetpunt voor de periode 1985 - 2014**, cfr. \@ref(h3-2:drempelwaarde), die periodiek, bijv. elke vijftien jaar, met meer recente data kan herijkt worden, cfr. \@ref(h2-2-1:referentietoestand).
+
+## De voorbereiding {#h4-1:voorbereiding}
+
+Tot op heden kon met het software-programma Menyanthes voor 58 meetpunten een tijdreeksmodel gemaakt worden, waarbij 
+
+- neerslag en potentiële evapotranspiratie voor minstens 66% van de waargenomen variatie kon verklaren en
+- er geen significante trend in de tijdreeks kon gedetecteerd worden.
+
+Met deze modellen werden 20 tijdreeksen van dagelijkse grondwaterpeilen gesimuleerd voor de periode 1985 - 2018.
+Deze data werden in een centrale databank opgeslagen om daarna te worden ingelezen in R. 
+De hierop volgende analysen gebeurden volledig in R.
+
+```{r functies}
+#grafiek voor het bekijken van de modelfitting. Het plot de gemeten data (in werkelijkheid omvatten ze zowel geïmputeerde data als veldmetingen) en de gefitte waarden incl credible interval. 
+
+#hulpfunctie voor het maken van minor ticks in de grafieken
+insert_minor <- function(major_labs, n_minor) {
+  labs <- c( sapply( major_labs, function(x) c(x, rep("", 4) ) ) )
+  labs[1:(length(labs) - n_minor)]}
+
+#plot van de modelfitting
+plotfitting <- function(indic_basis, respons, gemid, og, bg) {
+  og <- enquo(og)
+  bg <- enquo(bg)
+  # PI_data <- indic_basis %>% group_by(jaar) %>% 
+  #   summarise(ymax_data = max(!! bg, na.rm = TRUE), ymin_data = min(!! og, na.rm = TRUE))
+  
+  PI_data <- indic_basis %>% group_by(jaar, tijdsvenster) %>% 
+    summarise(ymax_data = max(!! bg, na.rm = TRUE), ymin_data = min(!! og, na.rm = TRUE)) %>% 
+    ungroup() %>% 
+    inner_join(dagen_droog_lang_breed)
+  
+  p <- ggplot(data = PI_data, aes_string(y = respons, x = "jaar"))
+  p <- p + xlab("Jaar") + ylab("Aantal dagen onder kritsche drempelwaarde")
+  # p <- p + theme(text = element_text(size = 15))
+  p <- p + geom_point(shape = 16, size = 2, col = "black")
+  p <- p + geom_line(aes_string(x = "jaar", y = gemid), size = 1, color = "red")
+  p <- p + geom_ribbon(aes(x = jaar, ymax = ymax_data, ymin = ymin_data),
+                       fill = grey(0.5),
+                       alpha = 0.4)
+  p <- p + scale_x_continuous(breaks = 1985:2020, 
+                              labels = insert_minor(seq(1985, 2020, by = 5), 4))
+  # p <- p + theme(strip.text = element_text(size = 15))
+  p <- p + facet_wrap(vars('tijdsvenster (jaren) ' = tijdsvenster), labeller = label_both)
+  return(p)
+}
+
+#plot van de rollende gemiddelden
+plotma.basis <- function(indic_finaal, indicator, intercept) {
+  p <- ggplot(data = indic_finaal ) + 
+    geom_line(data = indic_finaal %>% filter(jaar >= 1985 + tijdsvenster), aes(x = jaar, y = lcl), linetype = "longdash") +
```

Comment:
I think it can more efficient / readable if you create a new variable `period = jaar >= 1985 + tijdvenster`. Then you can add `alpha = period` to the `aes()`.

Reply:
This didn't work: Error: geom_path: If you are using dotted or dashed lines, colour, size and linetype must be constant over the line